app/questionnaire/questionnaire_manager.py

```diff
@@ -95,6 +105,8 @@ def go_to_state(self, item_id):
             self._create_new_state(item_id)
         StateManager.save_state(self)
 
+        StateManager.save_state(self)
```

Comment:
This is fixed in a later commit... These were cherry picked from the eq-525 branch to unblock warren, so I missed it.  Will fix now